subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaExecutableUtils.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.internal.deprecation.DeprecationLogger;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+
+public class JavaExecutableUtils {
+
+    public static File validateExecutable(String executable) {
```

Comment:
🤔 This method name is prefixed `validate`, same as the method below it, but not only performs some checks on the given input, it transforms it and returns a different object.  I think the name is a little misleading, I'd expect another `void` method here.  It's nitpicky, but I think this would be a little clearer as:

`public static void validateExecutable(File executable)` and let the caller create and pass the new `File` instance.